### PR TITLE
Integrate Mayhem

### PR DESCRIPTION
**Pull request by 4812571**
### Description
Integrated Mayhem, a fuzz testing solution by ForAllSecure

### Checklist
- [ ] Code compiles correctly
- [x] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [x] Extended the README / documentation, if necessary
- [x] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -31,6 +31,7 @@ Daniel Nichter <nil at codenode.com>
 Daniël van Eeden <git at myname.nl>
```
Comment (xansec) [inline AUTHORS:31]: Is adding your name to AUTHORS part of the contributing guidelines? I wouldn't consider Dockerfiles/Mayhemfiles core authorship.

File: mayhem/Mayhemfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Namespaced project name that the target belongs to
```
Comment (xansec) [inline mayhem/Mayhemfile]: I'd get rid of the commented lines here (especially the image).

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: AUTHORS (hunk 1 above)
Comment (4812571) [reply AUTHORS:31]: It appears to be, to quote the contributing guidelines:

> By contributing to this project, you share your code under the Mozilla Public License 2, as specified in the LICENSE file. Don't forget to add yourself to the AUTHORS file.

It's also part of the checklist to submit a commit. While I do agree that adding Dockerfiles/mayhemfiles to an already harnessed repository is not core authorship, I was trying to follow the guidelines as they were written.

What are your thoughts?

File: mayhem/Mayhemfile (hunk 2 above)
Comment (4812571) [reply mayhem/Mayhemfile]: I have addressed this.

File: AUTHORS (hunk 1 above)
Comment (xansec) [reply AUTHORS:31]: For now I'd still omit it. We actually haven't decided if we're fully going to push every repo to upstream, or if we're simply going to maintain them as forks under `mayhemheroes`. If the latter, then this code won't ever make it to upstream anyways. If we decide to merge later we can revisit.

**Review by xansec — APPROVED** [review]
On second thought, we can always remove this later. I'll go ahead and merge LGTM!

**Pull request merged**